### ISSUE TITLE
matchMedia.addListener is deprecated

### ISSUE THREAD

**Issue by alexgibson** [mozilla/protocol#854]
## Description

As per https://developer.mozilla.org/en-US/docs/Web/API/MediaQueryList/addListener, `addListener` is deprecated. Whilst most browsers alias it to the newer [`addEventListener`](https://developer.mozilla.org/en-US/docs/Web/API/MediaQueryListEvent), there's no guarantee that this will be true forever.

We should update where `addListener` is used in Protocol to use `addEventListener`, and then fallback to `addListener` for older browsers such as IE.

Rough feature detect (needs testing)

```javascript
if (window.matchMedia('all').addEventListener) {
    // use addEventListener('change')
else if (window.matchMedia('all').addListener) {
    // use addListener
}
```

**Comment by reemhamz** [comment]
Been looking to get my hands dirty with JavaScript. Picking this 🎟 up

**Comment by reemhamz** [comment]
Taking a look at caniuse.com, here are the comparisons between:
- [`addListener`](https://caniuse.com/?search=addlistener) vs 
- [`addEventListener`](https://caniuse.com/?search=addeventlistener)

Seems like `addEventListener` has even more support than `addListener`, and has been supporting IE for longer.
Based on that evaluation, I'm not sure we'd need the `addListener` fallback in our code?

**Comment by alexgibson** [comment]
addEventListener is not supported on matchMedia in IE. If you try it, you should see an error

**Comment by reemhamz** [comment]
Ah -- I missed that when I was reading up on the changes. Thanks Alex

**Comment by reemhamz** [comment]
Closed with #859
